cmd/vmwrapper/vmwrapper.go

```diff
@@ -75,13 +75,42 @@ func main() {
 	// for cgroups that aren't managed by libvirt.
 	// If we don't do this, the VM pod will be killed by kubelet when Virtlet pod
 	// is removed dnd cgroup-per-qos is enabled in kubelet settings.
+	podId := os.Getenv(config.VmPodIdEnvVarName)
+	glog.V(5).Infof("Got VM POD ID: %s", podId )
+
+	computeCgroupRoot := os.Getenv(config.VmCgroupRootEnvVarName)
+	glog.V(5).Infof("Got CgroupRoot: %s", computeCgroupRoot )
+
 	cm := cgroups.NewManager(os.Getpid(), nil)
-	for _, ctl := range []string{"hugetlb", "systemd", "pids"} {
-		if _, err := cm.GetProcessController(ctl); err == nil {
-			err = cm.MoveProcess(ctl, "/")
+
+	controllers, err := cm.GetProcessControllers()
+
+	if err != nil {
+		glog.Warningf("Failed to get cgroup controllers: %v", err)
+	} else {
+		for ctl, ctlPath := range controllers {
+			glog.V(5).Infof("Cgroup controller name %s, path %s", ctl, ctlPath)
+
+			cgPath := "/"
+			// Preserve the previous behavior to put VM process to root for systemd, hugetlb and pids controllers
+			// otherwise, put the process under the desired cgroup created for the POD
+			if ctl != "systemd" && ctl != "hugetlb" && ctl != "pids" {
+				if controller, err := cm.GetProcessController(ctl); err == nil {
+					cgPath = computeCgroupRoot + "pod" + podId
```

Comment:
Should the VM process not go under a container ID path under podXYZ? This will make it easier for Vertical Scaling which uses UpdateContainerResources under a containerID.

Reply:
You mean the VM process directly under the /kubepods/ or under /kubepods/podxxx/containerxxx/ ?

the podxxx cgroup was created by k8s cg manager i think the intention to add all containers within the same pod ( including the pause container ) so it can take pod as the whole unit. for VM type, we do have the vmId as the containerId and technically we can put it under there as well. we can take this offline to discuss further. we have another task/issue for 630 to refine resource management and controls for VM and we can make change there if needed.

Reply:
Discussed with Vinay. we will address this with 3.b in issue https://github.com/futurewei-cloud/arktos/issues/266 in a whole.

Reply:
+1 on use the same cgroup path scheme with container pods.  This will ease other controllers and tools.